t/42-screenshots.t

```diff
@@ -86,6 +95,52 @@ subtest 'screenshots are unique' => sub {
     is $whatever[1], undef, 'no second result';
 };
 
+subtest 'limiting screenshots splitted into multiple Minion jobs' => sub {
+    subtest 'test setup' => sub {
+        # create additional screenshots to get an ID range of [1; 206]
+        $screenshots->create({filename => 'test-' . $_, t_created => DateTime->now(time_zone => 'UTC')}) for (0 .. 200);
+        my ($min_id, $max_id) = $schema->storage->dbh->selectrow_array('select min(id), max(id) from screenshots');
+        is($min_id, 1,   'min ID');
+        is($max_id, 206, 'max ID');
+    };
+
+    # run a limit_results_and_logs job with customized batch parameters
+    run_gru_job($t->app, limit_results_and_logs => [{screenshots_per_batch => 20, batches_per_minion_job => 5}]);
+
+    # check whether "limit_results_and_logs" enqueues further "limit_screenshots" jobs
+    my $minion        = $t->app->minion;
+    my $enqueued_jobs = $minion->jobs({states => ['inactive'], tasks => ['limit_screenshots']});
+    my (@enqueued_job_ids, @enqueued_job_args);
+    while (my $info = $enqueued_jobs->next) {
+        push(@enqueued_job_ids,  $info->{id});
+        push(@enqueued_job_args, $info->{args});
+    }
+    @enqueued_job_args = sort { $a->[0]->{min_screenshot_id} <=> $b->[0]->{min_screenshot_id} } @enqueued_job_args;
+    is_deeply(
+        \@enqueued_job_args,
+        [
+            [{min_screenshot_id => 1,   max_screenshot_id => 100, screenshots_per_batch => 20}],
+            [{min_screenshot_id => 101, max_screenshot_id => 200, screenshots_per_batch => 20}],
+            [{min_screenshot_id => 201, max_screenshot_id => 206, screenshots_per_batch => 20}],
+        ],
+        'limit_screenshots tasks enqueued'
+    ) or diag explain \@enqueued_job_args;
+
+    # perform enqueued jobs
+    my $worker = $minion->worker->register;
+    combined_like(
+        sub { $worker->dequeue(0, {id => $_})->perform for @enqueued_job_ids },
+        qr/removing screenshot/,
+        'screenshots being removed'
+    );
+    $worker->unregister;
+
+    subtest 'enqueued sub jobs have actually removed the screenshots' => sub {
+        is($screenshots->search({filename => 'test-' . $_})->count, 0, "test screenshot $_ removed again")
+          for (000 .. 200);
```

Comment:
where does this magic number come from? Not sure how long the test takes right now but in my experience Test::More calls in a big for loop can take comparably long time. Within the scalability test we also used a different approach. Maybe the same is good here as well

Reply:
Number does seem a bit arbitrary, considering the max_id above is 206.

Reply:
The 200 corresponds to the screenshot name and is consistent with the loop where those screenshots are inserted. I have nevertheless added a better check instead which doesn't need that number.